lib/kinesis_client/stream/coordinator.ex

```diff
@@ -82,17 +86,22 @@ defmodule KinesisClient.Stream.Coordinator do
     {:reply, graph, s}
   end
 
-  @impl GenServer
-  def handle_cast(
-        {:close_shard, shard_id},
-        %{shard_ref_map: shards, stream_name: sn, app_name: app_name} = state
-      ) do
-    {ref, _} = Enum.find(shards, fn {_monitor_ref, in_shard_id} -> in_shard_id == shard_id end)
+  def handle_call({:append_shards, child_shards}, _from, state) do
+    shard_pid_map =
+      Enum.reduce(
+        child_shards,
+        state.shard_pid_map,
+        fn %{"ParentShards" => parent_shards, "ShardId" => shard_id}, acc ->
+          maybe_start_shard(parent_shards, shard_id, state, acc)
+        end
+      )
 
-    Process.demonitor(ref, :flush)
-    Shard.stop(Shard.name(app_name, sn, shard_id))
+    shard_map =
+      child_shards
+      |> Enum.group_by(&Map.fetch!(&1, "ShardId"))
```

Comment:
I think this should be:

```suggestion
      |> Map.new(&{Map.fetch!(&1, "ShardId"), &1})
```

Because in other locations we update this map expecting the shape to be `%{String.t() => map()}` but here we'll return `%{String.t() => [map()]}`.

Reply:
Sure. FWIW `shard_map` is not actually used anywhere 🫣 but I'm keeping it for now so it's easier to inspect if we need to.